### PR TITLE
Fix exon/intron boundary determine process and frameshift that affects initiation site

### PR DESCRIPTION
**Pull request by nokara26**
I fixed exon/intron boundary determine process to check position only changed.
And I fixed frameshift  that affects initiation site.

### CODE REVIEW

File: src/varity/vcf_to_hgvs/protein.clj [hunk 1]
```diff
@@ -430,8 +449,15 @@
                      format-alt-prot-seq
                      (subs (dec (+ ppos offset)))
                      (string/index-of "*"))]
-    (if (= alt \*)
+    (cond
+      (or ref-include-utr-ini-site-boundary
+          (not (string/starts-with? alt-prot-seq "M")))
```
Comment (nokara26) [inline src/varity/vcf_to_hgvs/protein.clj:454]: To exclude UTR and initiation site is changed but first amino acid of alt-prot-seq is "M".

File: src/varity/vcf_to_hgvs/protein.clj [hunk 2]
```diff
@@ -20,11 +20,14 @@
...
     (and (not (= 1 nref nalt))
          (not= 1 (count exon-ranges))
-         (some (fn [[s e]]
-                 (and (not= s e)
-                      (or (and (< pos s) (<= s (+ pos nref -1)))
-                          (and (<= pos e) (< e (+ pos nref -1))))))
-               exon-ranges))))
+         (let [[pos nref] (if (= (first ref) (first alt))
+                            [(inc pos) (dec nref)]
+                            [pos nref])]
```
Comment (nokara26) [inline src/varity/vcf_to_hgvs/protein.clj:25]: only check changed position

**Review by federkasten — COMMENTED** [review]
Thanks!

I noticed the `overlap-exon-intron-boundary?` function could use more versatility, so I created #93. I would like to merge this PR after #93.

I checked out the other changes in this PR. They look good to me! :ok_hand:

**Review by federkasten — APPROVED** [review]
Thanks for reviewing #93. Approved.

**Pull request merged**